va/wayland/va_wayland.c

```diff
@@ -54,15 +54,6 @@ va_wayland_error(const char *format, ...)
     va_end(args);
 }
 
-static int
-va_DisplayContextIsValid(VADisplayContextP pDisplayContext)
-{
-    VADriverContextP const pDriverContext = pDisplayContext->pDriverContext;
-
-    return (pDriverContext &&
-            pDriverContext->display_type == VA_DISPLAY_WAYLAND);
```

Comment:
this will not be checked?

Reply:
The display_type is set alongside the isValid hook, in the respective GetDisplay function. As such it will always be the same (barring very specific memory corruption, which is a straw man argument), is borderline nonsensical and can be removed.